public_profiler/v3.5-DD/ac.js

```diff
@@ -67,32 +67,20 @@ function ACChangeArmor()
     return;
 
   sheet().ACArmor.value  = 0;
-  if ( sheet().Armor1Worn.checked )
-  {
-      sheet().Armor1Carried.checked = true;
-      sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor1Bonus)));
-  }
-
-  if ( sheet().Armor3Worn.checked )
-  {
-      sheet().Armor3Carried.checked = true;
-      sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor3Bonus)));
-  }
-
-  if ( sheet().Armor4Worn.checked )
-  {
-      sheet().Armor4Carried.checked = true;
-          sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor4Bonus)));
-  }
-
   sheet().ACShield.value  = 0;
-  if ( sheet().Armor2Worn.checked )
-  {
-      sheet().Armor2Carried.checked = true;
-     sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
+
+  for (var i = 1; i <= 8; i++) {
+      if (sheet()["Armor" + i + "Worn"].checked) {
+          sheet()["Armor" + i + "Carried"].checked = true;
+          if (i == 2)
```

Comment:
With 8 slots, it might be useful to have two slots that count as shield, on the theoretical possibility that someone might have two shields and swap between them.

Reply:
Fair enough - I even considered the possibility of adding a second but passed thinking "why, though ?" - which you've answered.  I guess i % 4 == 2 will do as test ...